index.ts

```diff
@@ -54,28 +126,18 @@ interface KoaLike {
  */
 export async function patchInstance(
   instance: AxiosInstance,
-  app: RequestListener | KoaLike,
+  app: Application,
 ): Promise<AxiosTestInstance> {
-  const server = createServer(app instanceof Function ? app : app.callback());
-  await new Promise((resolve) => {
-    server.listen(undefined, '127.0.0.1', resolve);
-  });
-  const { address, port } = server.address() as AddressInfo;
+  const { close, uri } = await startServer(app);
   const inst = instance as AxiosTestInstance;
   const { baseURL } = instance.defaults;
-  inst.defaults.baseURL = String(new URL(baseURL || '', `http://${address}:${port}`));
-  inst.close = (): Promise<void> =>
-    new Promise((resolve, reject) => {
-      inst.defaults.baseURL = baseURL;
-      inst.close = (): Promise<void> => Promise.resolve();
-      server.close((error) => {
-        if (error) {
-          reject(error);
-        } else {
-          resolve();
-        }
-      });
-    });
+  inst.defaults.baseURL = String(new URL(baseURL || '', uri));
+  inst.close = async (): Promise<void> => {
+    inst.defaults.baseURL = baseURL;
+    await close();
+    await new Promise((resolve) => setTimeout(() => resolve(), 3e3));
```

Comment:
why is this 3 second timeout here? It makes every test take 3s extra

Reply:
I’m not sure how that got in. I removed it and made a patch release.